Compilation Failure - empty scalar initializer - Inheritable objects & Atomics [OA]RC on DEVEL

The following fails to compile with the listed error using ARC or ORC however compiles successfully on REFC

```nim
import atomics

type
  # Deprivation* = ref object of RootObj    # compiles
  Deprivation* = object of RootObj        # fails
    memes*: Atomic[int]
  Zoomer = object
    dopamine: Deprivation                 # fails
    # dopamine: ref Deprivation             # compiles

# discard Deprivation()                     # compiles
discard Zoomer()                          # fails
```

I have tagged which lines I have narrowed down to causing issues and what doesn't.

This was found while trying to use elijahrs lockfreequeue types as fields of objects.

```
C:\Users\PC\nimcache\cpsloop_r\@mcpsloop.nim.c:199:132: error: empty scalar initializer
 static NIM_CONST tyObject_Zoomer__yhOZMQHEaznD9aq8IQHJaMQ TM__8CKjD9bWE6mTvTNPLW3KYWw_2 = {{{(&NTIv2__DdWjNMQNgJ9aHiVjMo4gjYw_)}, {{}}}}

                        ^
C:\Users\PC\nimcache\cpsloop_r\@mcpsloop.nim.c:199:132: note: (near initialization for 'TM__8CKjD9bWE6mTvTNPLW3KYWw_2.dopamine.memes.value')
Error: execution of an external compiler program 'gcc.exe -c  -w -fmax-errors=3 -mno-ms-bitfields -O3 -fno-strict-aliasing -fno-ident   -IC:\Users\PC\.choosenim\toolchains\nim-#devel\lib -Ic:\Users\PC\Documents\Programming\ParthianProject\Modules\cps_minfosclient\cpsbridge -o C:\Users\PC\nimcache\cpsloop_r\@mcpsloop.nim.c.o C:\Users\PC\nimcache\cpsloop_r\@mcpsloop.nim.c' failed with exit code: 1
```

--- a/lib/pure/concurrency/atomics.nim
+++ b/lib/pure/concurrency/atomics.nim
@@ -288,10 +288,10 @@ else:
     type
       # Atomic* {.importcpp: "_Atomic('0)".} [T] = object
 
-      AtomicInt8 {.importc: "_Atomic NI8", size: 1.} = object
-      AtomicInt16 {.importc: "_Atomic NI16", size: 2.} = object
-      AtomicInt32 {.importc: "_Atomic NI32", size: 4.} = object
-      AtomicInt64 {.importc: "_Atomic NI64", size: 8.} = object
+      AtomicInt8 {.importc: "_Atomic NI8".} = int8
+      AtomicInt16 {.importc: "_Atomic NI16".} = int16
+      AtomicInt32 {.importc: "_Atomic NI32".} = int32
+      AtomicInt64 {.importc: "_Atomic NI64".} = int64
 
     template atomicType*(T: typedesc[Trivial]): untyped =
       # Maps the size of a trivial type to it's internal atomic type

--- a/tests/stdlib/concurrency/tatomics.nim
+++ b/tests/stdlib/concurrency/tatomics.nim
@@ -1,6 +1,6 @@
 # test atomic operations
 
-import atomics, bitops
+import std/[atomics, bitops]
 
 type
   Object = object
@@ -607,3 +607,23 @@ block clear:
   doAssert not location.testAndSet
   location.clear(moRelease)
   doAssert not location.testAndSet
+
+block: # bug #18844
+  when not defined(cpp): # cpp pending pr #18836
+    type
+      Deprivation = object of RootObj
+        memes: Atomic[int]
+      Zoomer = object
+        dopamine: Deprivation
+
+    block:
+      var x = Deprivation()
+      var y = Zoomer()
+      doAssert x.memes.load == 0
+      doAssert y.dopamine.memes.load == 0
+
+    block:
+      var x: Deprivation
+      var y: Zoomer
+      doAssert x.memes.load == 0
+      doAssert y.dopamine.memes.load == 0

